### PR TITLE
New tests for gradient interpolation

### PR DESCRIPTION
**Pull request by faceless2**
Followup from https://github.com/web-platform-tests/wpt/pull/34673

WPT testing works as expected[1] so this PR

* Fixes the reftest for gradient-eval-001 to use the correct RGB value
* Adjusts gradient-eval-002 and adds several more tests for gradient interpolation as of the current draft of css-images

1.  https://wpt.fyi/results/css/css-images/gradient/gradient-eval-001.tentative.html?sha=f443641477&label=master&max-count=1 - which would have passed had I used the correct RGB value in the reftest.

### CODE REVIEW

**Review by tabatkins — APPROVED** [review]

**Comment by nt1m** [comment]
@faceless2 Is this ready to merge?

**Comment by faceless2** [comment]
Yes it is ready, thanks Tim.

File: css/css-images/gradient/gradient-eval-004-ref.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<!DOCTYPE html>
+<html>
+ <head>
+  <title>Gradient interpolation</title>
+  <link rel="author" title="Mike Bremford" href="mailto:mike@bfo.com">
+  <style>
+   .test {
+       width: 100px;
+       height: 100px;
+       background: rgb(0% 50% 50%);
```
Comment (yisibl) [inline css/css-images/gradient/gradient-eval-004-ref.html:10]: I'm afraid this is not the correct expected value, gradient interpolation is different from `color-mix()`. It's not equivalent to `color-mix( in srgb, rgb(0% 0% 50%), rgb(0% 100% 50%) )`.

@svgeesus WDYT?

**Comment by faceless2** [comment]
This PR is almost three years old now and predates the "interpolate in OKLab by default" decision, so the mid-points will need re-evaluating. I will update the PR files to reflect the spec when I get a moment over the next few days.

**Pull request merged**